src/Modules/UngradedModules.jl

```diff
@@ -1539,8 +1543,17 @@ function is_canonically_isomorphic(M::SubModuleOfFreeModule, N::SubModuleOfFreeM
   return SubModuleOfFreeModule(G, [f(v) for v in gens(M)]) == N
 end
 
-#+(M::SubModuleOfFreeModule, N::SubModuleOfFreeModule) = sum(M, N)
+Base.:+(M::SubModuleOfFreeModule, N::SubModuleOfFreeModule) = sum(M, N)
+
+function Base.:*(I::MPolyIdeal{T}, M::SubModuleOfFreeModule{T}) where {T<: MPolyRingElem}
```

Comment:
I thought these methods existed already. The failure of the tests suggests that the this overrides an existing generic method, but with incompatible output format. Return a map also?

Reply:
I renamed my method to solve the problem .